### PR TITLE
Add additional search view pages to the Nuxt app

### LINKED ISSUE #2774: Create a page for tag/creator/source views
## Problem

We need a new page for the Additional search views projects. We should handle the following paths:
`/pages/_mediaType/tag/_tag`
`/pages/_mediaType/source/_source`
`/pages/_mediaType/source/_source/creator/_creator` (this page might not be needed as it might be handled by the source page)
## Description

To make sure that the mediaType, source and creator parameters are valid, this page should use the [validate method](https://v2.nuxt.com/docs/components-glossary/validate/) to make sure that and show an error page if necessary.
```typescript
function validate({ params, $pinia }): boolean {
  const { collection, mediaType, term } = params
  // Check that collection is one of ["tag", "creator" or "source"],
  // and mediaType is one of `supportedMediaTypes`.
  // Check that `term` is correctly escaped.
  // If the params are not valid, return `false` to show the error page.
  return isValid ? true : false
}
```
This page should also update the state (`searchType`, `searchTerm` and `searchBy` and `provider` filters) in the search store and handle fetching using mediaStore’s `fetchMedia` method in the useFetch hook.

Since it is not possible to change the path or query parameters from this page client-side, fetching can be much simpler than on the current search page (that has to watch for changes in the route and fetch if necessary).

This page should use `VCollectionHeader` and the image grid or the audio collection.

### Dependencies
Issue is blocked on the following dependencies:
- [x] #2694
- [x] #2729 
- [x] #2858

### LINKED ISSUE #2858: Update the Nuxt store to allow for searches by source/creator/tag
## Problem

The Nuxt app store and URL construction functions need to be updated to allow for getting the tag/source/creator collections.

## Description

The `searchBy` filter in the search store should be updated with the new values..

The API request URL is constructed from the search store state in the [`prepare-search-query-params` method](https://github.com/WordPress/openverse/blob/b4b46b903731870015c475d2c08eebef7ec6b25b/frontend/src/utils/prepare-search-query-params.ts#L22-L25).
This method should be updated to use the `searchBy` filter value to construct the API request path as described in the "API Changes" section.

## Additional context

Additional search views Implementation plan: https://docs.openverse.org/projects/proposals/additional_search_views/20230719-implementation_plan_additional_search_views.html

### PR DESCRIPTION
**Pull request by obulat**
~~This PR is based on #3136, and will be undrafted after that PR is merged.~~

## Fixes

Fixes #2858 by @obulat
Fixes #2774 by @obulat 

## Description

With this PR, you can view the tag/source/creator collections in the Nuxt app.

To show the collections, we add `_tag.vue` and `_source.vue` pages to both `image` and `audio`, with dynamic routes. These pages check the path to make sure that the path has the necessary parameters: tag name for the tag page; an existing source for source/creator pages; `/creator/` path part and the creator name for the creator pages. 

In the first iteration, I used a single page with a fully-dynamic path (`/_mediaType/_collection/_.vue`) that would then validate the `mediaType`, `collection` and the collection parameters. However, because `_mediaType` can accept any string path, it would also match the locale routes such as `/ar/non_existent_page`), so I split this single page into 4.

The collection pages have the collection headers (using `VCollectionHeader` from #3365) and either a `VAudioCollection` or `VImageGrid`, as well as the "Load more" button. When you navigate from the collection page to a single result page, you also get the "Back to results" link at the top that goes back to the collection page.

This PR adds a `strategy` and `collectionParams` props to the `search` store. Whenever we set the `searchTerm` (which is only available for search, not collection), we reset the `strategy` to `default` and nullify the `collectionParams`.

The API service had to be updated to accept a `slug` for paths like `/tag/tag_name`, `/source/source_name`, `/source/source_name/creator/creator_name`, which are added to the `/<mediaType>` path.
The media store was updated to use the necessary slug from the search store `collectionParams`.

## Testing Instructions

Run the app using `just frontend/run dev`. Go to `/preferences` page and turn on the `additional_search_views` feature flag.

The easiest way to land on a collection page is from the single result page. Click on the source, creator or tag buttons - and you will be taken to the corresponding page.
Check that in non-English locales, the links have the correct locale path, and when navigating, the locale is kept. 
Check that if you go from a collection page to a single result page, you can go back to the collection page using "Back to results" page.
Check that if you go back to homepage from the collection page, you can search for something.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/components/VLoadMore.vue [hunk 1]
```diff
@@ -78,6 +86,14 @@ export default defineComponent({
...
       })
 
+      router.push({
+        path: route.value.path,
+        query: {
+          ...route.value.query,
+          page: `${currentPage.value + 1}`,
+        },
+      })
+
```
Comment (obulat) [inline frontend/src/components/VLoadMore.vue:96]:
```suggestion
```

File: frontend/src/components/VLoadMore.vue [hunk 2]
```diff
@@ -18,7 +18,7 @@ import { computed, defineComponent, onMounted, ref, watch } from "vue"
 import { storeToRefs } from "pinia"
 import { useElementVisibility } from "@vueuse/core"
 
-import { useRoute } from "@nuxtjs/composition-api"
+import { useRoute, useRouter } from "@nuxtjs/composition-api"
```
Comment (obulat) [inline frontend/src/components/VLoadMore.vue:21]:
```suggestion
import { useRoute } from "@nuxtjs/composition-api"
```

**Review by fcoveram — CHANGES_REQUESTED** [review]
It works great 👏 

I just noticed one area of improvement:
* When `VCollectionHeader` has a padding of 40px on the sides (`xs / sm / md`), top padding looks better with 8px.
* When `VCollectionHeader` has a padding of 24px on the sides (`lg / xl / 2xl`), top padding looks better with 0px.

On the other hand and something out of this scope, I noticed that loading these views keeps the page empty for a few milliseconds. It seems good idea showing the [media's skeleton](https://docs.openverse.org/storybook/?path=/story/components-skeleton--all-tab). Here is a video recording of what I see.

https://github.com/WordPress/openverse/assets/895819/bbb35dc6-2a85-4f28-aa0f-275d9cb8b40b

If you think is a good idea to include. I can request it in a new ticket.

**Review by dhruvkb — APPROVED** [review]
The pages work well, when they do work. I have nothing to add to @fcoveram's review, LGTM from my side!

**Review by AetherUnbound — COMMENTED** [review]
THIS IS SO COOOOOL! 🤩 It's amazing to see this come together on the frontend!! I have a few questions, but also - should we have any snapshots as part of this PR too?

**Review by AetherUnbound — APPROVED** [review]
I don't actually have anything additional that's blocking 😄

**Comment by obulat** [comment]
> On the other hand and something out of this scope, I noticed that loading these views keeps the page empty for a few milliseconds. It seems good idea showing the [media's skeleton](https://docs.openverse.org/storybook/?path=/story/components-skeleton--all-tab). Here is a video recording of what I see.

I updated the paddings, @fcoveram . As for the loading skeletons: they were broken on `main` branch because a variable was renamed. I fixed the error on `main` and rebased this branch, so now you can see the loading skeletons when the media is loading.

File: frontend/src/utils/validate-collection-params.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import { warn } from "~/utils/console"
+import { useFeatureFlagStore } from "~/stores/feature-flag"
+
+export function validateCollectionParams({
+  firstParam,
+  mediaType,
+  params,
+  $pinia,
+}: {
+  firstParam: "tag" | "source"
```
[the remark this answers is not in the document]
Comment (AetherUnbound) [reply frontend/src/utils/validate-collection-params.ts:32]: Got it, hmm. I'm not sure I have a better term 😅 I think I also missed while looking at this that `creator` is the second part.

**Review by fcoveram — CHANGES_REQUESTED** [review]
I noticed one odd spacing in `VCollectionHeader` that can look better with the following two changes:
* Audio results of source, tag, and author in all breakpoints, margin-bottom looks better with `8px`. I refer to [this area](https://share.cleanshot.com/ZC2FXxlm).
* Image results of source, tag, and author in all breakpoints, margin-bottom looks better with `16px`. I refer to [this area](https://share.cleanshot.com/ZKY9kd3D).

Pardon the picky territory my suggestions belong to, but QA is the perfect stage for adjustments.

**Comment by obulat** [comment]
> I noticed one odd spacing in `VCollectionHeader` that can look better with the following two changes:
> 
> * Audio results of source, tag, and author in all breakpoints, margin-bottom looks better with `8px`. I refer to [this area](https://share.cleanshot.com/ZC2FXxlm).
> * Image results of source, tag, and author in all breakpoints, margin-bottom looks better with `16px`. I refer to [this area](https://share.cleanshot.com/ZKY9kd3D).
> 
> Pardon the picky territory my suggestions belong to, but QA is the perfect stage for adjustments.

Thank you for re-reviewing! I've updated the paddings.

I also noticed that the collection pages really didn't render on the server at all, not just because of some local problems. Thank you for flagging that, @fcoveram. I fixed the logic of the collection page parameter validation, and now they do render on the server.
I've added the e2e tests, but didn't add visual regression tests to not make the tests run too long.

**Review by fcoveram — APPROVED** [review]
It looks great ✨🚀

**Pull request merged**